bridge/svix-bridge-plugin-kafka/src/input.rs

```diff
@@ -0,0 +1,214 @@
+use std::{
+    str,
+    time::{Duration, Instant},
+};
+
+use rdkafka::{
+    consumer::{CommitMode, Consumer as _},
+    error::KafkaError,
+    Message as _,
+};
+use svix_bridge_types::{
+    async_trait, svix::api::Svix, CreateMessageRequest, JsObject, SenderInput, SenderOutputOpts,
+    TransformationConfig, TransformerInput, TransformerInputFormat, TransformerJob,
+    TransformerOutput, TransformerTx,
+};
+use tokio::task::spawn_blocking;
+
+use crate::{config::KafkaInputOpts, Error, Result};
+
+pub struct KafkaConsumer {
+    name: String,
+    opts: KafkaInputOpts,
+    transformation: Option<TransformationConfig>,
+    transformer_tx: Option<TransformerTx>,
+    svix_client: Svix,
+}
+
+impl KafkaConsumer {
+    pub fn new(
+        name: String,
+        opts: KafkaInputOpts,
+        transformation: Option<TransformationConfig>,
+        output: SenderOutputOpts,
+    ) -> Result<Self> {
+        Ok(Self {
+            name,
+            transformation,
+            transformer_tx: None,
+            opts,
+            svix_client: match output {
+                SenderOutputOpts::Svix(output) => {
+                    Svix::new(output.token, output.options.map(Into::into))
+                }
+            },
+        })
+    }
+
+    #[tracing::instrument(skip_all)]
+    async fn process(&self, msg: &rdkafka::message::BorrowedMessage<'_>) -> Result<()> {
+        let payload = msg.payload().ok_or_else(|| Error::MissingPayload)?;
+        let payload = if let Some(transformation) = &self.transformation {
+            let input = match transformation.format() {
+                TransformerInputFormat::Json => {
+                    let json_payload =
+                        serde_json::from_slice(payload).map_err(Error::Deserialization)?;
+                    TransformerInput::JSON(json_payload)
+                }
+                TransformerInputFormat::String => {
+                    let raw_payload = str::from_utf8(payload).map_err(Error::NonUtf8Payload)?;
+                    TransformerInput::String(raw_payload.to_string())
+                }
+            };
+
+            let script = transformation.source().clone();
+            let object = self.transform(script, input).await?;
+            serde_json::from_value(serde_json::Value::Object(object))
+                .map_err(Error::Deserialization)?
+        } else {
+            serde_json::from_slice(payload).map_err(Error::Deserialization)?
+        };
+
+        let CreateMessageRequest {
+            app_id,
+            message,
+            post_options,
+        } = payload;
+
+        // FIXME(jplatte): I think we should always set the idempotency key
```

Comment:
So about this, do you agree @svix-gabriel? If so I'll update the PR. I'm unclear why the `CreateMessageRequest` in the queue would itself set the idempotency key, but it is part of `post_options` there. I was thinking we should always set it to something like `svix_bridge_kafka_{topic}_{stream_position}` such that a failure to commit and retry, or even a crash and restart after the webhook was sent doesn't re-send it.

Reply:
To be honest, I'm not really sure. Your suggestion sounds reasonable to me.

I know @svix-onelson is much more familiar w/bridge and how it's used than I am, so he might have some insight.